rfcs/008-map-indexes.md

```diff
@@ -0,0 +1,202 @@
+"SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this
+document are to be interpreted as described in
+[RFC 2119](https://www.rfc-editor.org/rfc/rfc2119.txt).
+
+## Terminology
+
+`Sequence`: a 13-byte value formed by combining the current `Incarnation` of the database and the `Versionstamp` of the transaction. Sequences are monotonically increasing even when a database is relocated across FoundationDB clusters. See (RFC002)[LINK TBD] for a full explanation.
+
+`View Signature`: A md5 hash of the views, options, view language defined in a design document.
+
+---
+
+## Detailed Description
+
+CouchDB views are used to create secondary indexes in a database. An index is defined by creating map/reduce functions in a design document. This document describes building the map indexes on top of FoundationDB (FDB).
+
+An example map function is shown below:
+
+```json
+{
+  "_id": "_design/design-doc-id",
+  "_rev": "1-8d361a23b4cb8e213f0868ea3d2742c2",
+  "views": {
+    "map-view": {
+      "map": "function (doc) {\n  emit(doc._id, 1);\n}"
+    }
+  },
+  "language": "javascript"
+}
+```
+
+### Data model
+
+The data model for a map indexed is:
+
+```
+(<database>, ?DB_VIEWS, <view_signature>, ?VIEW_UPDATE_SEQ) = Sequence
+{<database>, ?DB_VIEWS, <view_signature>, ?VIEW_ID_INFO, view_id, ?VIEW_ROW_COUNT} = <row_count>
+{<database>, ?DB_VIEWS, <view_signature>, ?VIEW_ID_INFO, view_id, ?VIEW_KV_SIZE} = <kv_size>
+(<database>, ?DB_VIEWS, <view_signature>, ?VIEW_ID_RANGE, <_id>, <view_id>) = [total_keys, total_size, unique_keys]
+(<database>, ?DB_VIEWS, <view_signature>, ?VIEW_MAP_RANGE, <view_id>, <key>, <_id>, ?ROW_KEYS <count>) = <emitted_keys>
+(<database>, ?DB_VIEWS, <view_signature>, ?VIEW_MAP_RANGE, <view_id>, {<key>, <_id>}, <count>, ?ROW_VALUE) = <emitted_value>
+```
+
+Each field is defined as:
+
+- `<database>` is the specific database namespace
+- `?DB_VIEWS` is the views namespace.
+- `view_signature` is the design documents `View Signature`
+- `?VIEW_UPDATE_SEQ` is the change sequence namespace
+- `?VIEW_ID_RANGE` is the map id index namespace
+- `?VIEW_MAP_RANGE` is the map namespace
+- `?ROW_KEYS` is the namespace for a row where the value is the emitted keys from a map function
+- `?ROW_VALUE` is the namespace for a row where the value is the emitted value from a map function
+- `_id` is the document id
+- `view_id` id of a view defined in the design document
+- `key` is the encoded emitted row key from a map function
+- `count` is a value that is incremented to allow duplicate keys to be emitted for a document
+- `emitted_key` is the emitted key from the map function
+- `emitted_value` is the emitted value from the map function
+- `row_count` number of rows in the index
+- `kv_size>` size of the index
+- `total_keys` is the number of keys emitted by a document
+- `total_size` is the size of the key/values emitted by the document
+- `unique_keys` is the unique keys emitted by the document
```

Comment:
is this the value of the unique keys or the count?

Reply:
Its a list of all the unqiue keys